src/linux/linux_io.c

```diff
@@ -110,17 +109,34 @@ static int prepare_peer_socket(int fd)
 {
 	static const int tcp_nodelay_on = 1;
 
-	if ((set_fd_non_blocking(fd) < 0) ||
-	    (setsockopt(fd, IPPROTO_TCP, TCP_NODELAY, &tcp_nodelay_on,
-	                sizeof(tcp_nodelay_on)) < 0)) {
+	if (set_fd_non_blocking(fd) < 0) {
 		log_err("Could not set socket to nonblocking!\n");
 		close(fd);
 		return -1;
 	}
 
-	if (configure_keepalive(fd) < 0) {
-		log_err("Could not configure keepalive!\n");
-		close(fd);
+	int domain;
+	socklen_t len;
+	if (getsockopt(fd, SOL_SOCKET, SO_DOMAIN, &domain, &len) < 0) {
+		log_err("could not determine socket domain");
+		return -1;
+	}
+	if ((domain==AF_INET) || (domain==AF_INET6)) {
```

Comment:
Please adhere to coding style so space around ==

Reply:
done